### DISCUSSION TITLE
I2C slave boards

### DISCUSSION THREAD

**Discussion by vipercortex** [alanbjohnston/CubeSatSim#274] [Q&A]
The MAIN board layout is designed to hold 8 INA219 current sensor breakout boards (4 on each side) spread over 2 I2C buses.

1) What is the purpose of having 8 current sensor boards? I understand you would like to have 1 for monitoring the batteries.

2) Is it possible to substitute these INA219 boards for other I2C slave boards (e.g. ADCS sensors)

3) Most other boards do not have the same pinout as the INA219. What would you recommend on how to mate the I2C interface of other boards to the MAIN board? Are there any convenient connectors that you know of to connect 

4) If the other I2C boards do not fit conveniently on the MAIN board, what would be the best approach to integrate them in an AMSAT cubesat assembly (e.g. on a separate board and connect via wires, ...)

**Comment by alanbjohnston** [comment]
Hi,

1.  The board monitors independently all 6 sides plus the battery and the bus voltage and current.
2.  You can certainly wire additional I2C sensors.  As long as there aren't address conflicts, and the pull-up resistance is maintained, it should work.
3. I would add additional sensors to the STEM Payload board.  If you add them to the Main board, they are on the Raspberry Pi Zero I2C interface, and you will need to modify the main code to include them.  If you add them to the STEM Payload board, they are on the Pro Micro I2C interface, so you just need to modify that code, which is simpler.
4. The prototyping area of the STEM Payload board  can be used for soldering your own sensors.  Or, you could add a board into the board stack with your sensors, as long as board stack still fits inside the frame, unless you change to a 2U frame which will have lots of room.

Good luck!
